bob-backend/src/pearl/group.rs

```diff
@@ -307,12 +317,23 @@ impl Group {
         let mut exist = vec![false; keys.len()];
         let holders = self.holders.read().await;
         for (ind, &key) in keys.iter().enumerate() {
+            let mut max_delete_ts = 0; // In case of multiple holders with same ts and delete action
+            let mut max_ts = 0;
```

Comment:
Why are timestamps separated? We want the latest one, whether it's been deleted or not

Reply:
The record does not exist if it was deleted

Reply:
This approach may, in corner cases, produce a different result than `get`. If 2 holders contain records with the same timestamp (the last one with the actual data and the previous one with the deletion), then `exist` will return that the record has been deleted, while `get` return the record as it is presented (which is the correct).
I suggest to synchronize `exist` and `get` logic and track only latest `timestamp` + `bool` flag for record presence